nested/azuresearch-none.json

```diff
@@ -0,0 +1,30 @@
+{
+    "$schema": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "moodleCommon": {
+            "metadata": {
+                "description": "Common Moodle values"
+            },
+            "type": "object"
+        }
+    },
+    "resources": [
+    ],
+    "variables": {
+        "documentation01": "This sub-template drives the Azure Search which is used as the access-point for other moodle VM's ",
+        "documentation02": "It expects certain values in the 'common' datastructure.",
+        "documentation03": "    azureSearchName                 - the name of the Azure Search.",
+        "documentation04": "    azureSearchSku                  - the level of Azure Search service.",
+        "documentation06": "    azureSearchReplicaCount         - number of Azure Search replicas.",
+        "documentation07": "    azureSearchPartitionCount       - number of Azure Search partitions.",
+        "documentation08": "    azureSearchHostingMode          - the type of Azure Search hosting mode.",
+        "azureSearchServiceId": "[resourceId('Microsoft.Search/searchServices', parameters('moodleCommon').azureSearchName)]"
```

Comment:
This is strange. As far as I understand about ARM, this line should fail if azureSearchSwitch param is false (because the resource is not deployed at all, and calling resourceId() function on a nonexistent resource causes an ARM deployment failure). Did you test the azureSearchSwitch==false case at all? Please do make sure to test the false case as well.

The only reason why I had to use this trick (...-none.json) is because the ARM if(,,) function appears to be eager (call-by-value), not lazy, so it evaluates the listKey() even when the cond in if(cond,listKey(),'None') is false. I thought you'd be following the same pattern here, but apparently not. In this ...-none.json template, there should be no resourceId() function calls and no list*() function calls, to avoid that ARM restriction. Please let me know if this is unclear.